gui/control.js

```diff
@@ -18,61 +18,91 @@ window.addEventListener('unload', beforeQuit)
 window.addEventListener('DOMContentLoaded', main)
 
 
+const loadingMessage = document.getElementById('loading-message');
+const loadingOverlay = document.getElementById('loading-overlay');
+const showOverlay =false;
+
+function setLoadingOverlayVisibility(showOverlay) {
+     loadingOverlay.style.display = showOverlay ? 'flex' : 'none';
+}
 async function addButtonPressed() {
+  try {
+    setLoadingOverlayVisibility(true); // Show loading overlay before the action starts
     const nickname = await showTextInputPopup("Enter Nickname", "Add Account");
     if (!nickname) return;
     const error = await PSS.addAccount(nickname);
     if (!error) return;
-    showToast("Error!", errorToMessage[error])
+    showToast("Error!", errorToMessage[error]);
+  } finally {
+     setLoadingOverlayVisibility();// Hide loading overlay after the action completes (whether successful or not)
+  }
 }
 function openLinkInDefaultBrowser() {
-    shell.openExternal(' https://definex.in/discord');
+  shell.openExternal(' https://definex.in/discord');
 }
-function render() {
-    logger.debug("rendering");
-
-    accountsContainer.innerHTML = "";
-    const accountsDiv = document.createElement('div');
-    accountsDiv.classList.add('accounts-list');
-    accountsContainer.appendChild(accountsDiv);
-
-    const { accounts, currentUser } = PSS.returnAccountList();
-    logger.debug(JSON.stringify(accounts));
-    for (const nickname of accounts) {
-        const userCardString = templates.generate_user_card(nickname, nickname === currentUser);
-        accountsDiv.insertAdjacentHTML('beforeend', userCardString);
-
-        // Deleting User Profile
-        document.getElementById(`switch-btn-${nickname}`).addEventListener('click', async function () {
-            logger.debug(`switching ${nickname}`)
-            const error = await PSS.switchAccount(nickname);
-            if (!error) return;
-            showToast("Error!", errorToMessage[error])
-        });
-
-        document.getElementById(`delete-btn-${nickname}`).addEventListener('click', async function () {
-            const agreed = await showYesNoPopup(`Are you sure you want to delete ${nickname} account?`, "Cancel", "Delete Account");
-            if (!agreed) return;
-            logger.debug(`deleting ${nickname}`)
-            const error = await PSS.deleteAccount(nickname);
-            if (!error) return;
-            showToast("Error!", errorToMessage[error])
-        });
-    }
+
+
+async function switchaccounts(nickname) {
+  try {
+    setLoadingOverlayVisibility(true); // Show loading overlay before the action starts
+    logger.debug(`switching ${nickname}`);
+    const error = await PSS.switchAccount(nickname);
+    if (!error) return;
+    showToast("Error!", errorToMessage[error]);
+  } finally {
+    setLoadingOverlayVisibility(); // Hide loading overlay after the action completes (whether successful or not)
+  }
 }
```

Comment:
pls make sure only ur changes are commited, else it makes it very difficult to review.